### ISSUE TITLE
Problem exporting file to snappshot

### ISSUE THREAD

**Issue by tobiasK2001** [Open-Systems-Pharmacology/PK-Sim#1577]
Exported file to snappshot:
The warning showed up
![image](https://user-images.githubusercontent.com/25585248/97153027-1e7ab180-1772-11eb-912e-0a906e1ec026.png)

clicked OK to continue,
than it stopped with this:

Application:
PK-Sim® 8.0.22

Object reference not set to an instance of an object.

Stack trace:
```
at PKSim.Core.Snapshots.Mappers.SimulationMapper.<>c.<usedObervedDataFrom>b__31_0(DataRepository x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.usedObervedDataFrom(Simulation simulation, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.<MapToSnapshot>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<mapSimulationsToSnapshots>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<MapToSnapshot>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.<MapToSnapshot>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<exportSnapshotFor>d__21`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<ExportModelToSnapshot>d__6`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at OSPSuite.Core.Extensions.TaskExtensions.<SecureAwait>d__2`1.MoveNext()
```

**Comment by tobiasK2001** [comment]
I deleted the files with the red warning as I don't need them any more.
Than I doublecked that all Simulation are green. That every PI and SA can be opened. Safed and closed the project. After reopening I tried again. This time the warning did not show up, as no simulation was in changed state.
I could see the export was staring but than after a few minutes the oops error popped up again with this text:
Application:
PK-Sim® 8.0.22

Object reference not set to an instance of an object.

Stack trace:
```
at PKSim.Core.Snapshots.Mappers.SimulationMapper.<>c.<usedObervedDataFrom>b__31_0(DataRepository x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.usedObervedDataFrom(Simulation simulation, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.<MapToSnapshot>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<mapSimulationsToSnapshots>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<MapToSnapshot>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.<MapToSnapshot>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<exportSnapshotFor>d__21`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<ExportModelToSnapshot>d__6`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at OSPSuite.Core.Extensions.TaskExtensions.<SecureAwait>d__2`1.MoveNext()
```

**Comment by msevestre** [comment]
@tobiasK2001 Can you try with version 9.1 and see if the issue is fixed? You can simply install the portable version and check it out.

**Comment by msevestre** [comment]
It looks like an issue with some observed data there are referenced in a simulation but not available in the project or sthg like this

**Comment by tobiasK2001** [comment]
OK, It tried with the 9.1.2 portable and this is what I get:

Application:
PK-Sim® 9.1.2

Object reference not set to an instance of an object.

Stack trace:
```
at PKSim.Core.Snapshots.Mappers.SimulationMapper.<>c.<usedObservedDataFrom>b__33_0(DataRepository x)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.usedObservedDataFrom(Simulation simulation, PKSimProject project)
   at PKSim.Core.Snapshots.Mappers.SimulationMapper.<MapToSnapshot>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBaseExtensions.<MapTo>d__9`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<mapSimulationsToSnapshots>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.ProjectMapper.<MapToSnapshot>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Mappers.SnapshotMapperBase`2.<MapToSnapshot>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<exportSnapshotFor>d__21`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at PKSim.Core.Snapshots.Services.SnapshotTask.<ExportModelToSnapshot>d__6`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at OSPSuite.Core.Extensions.TaskExtensions.<SecureAwait>d__2`1.MoveNext()
```

**Comment by tobiasK2001** [comment]
Is it somehow possible to identify the Simulation or dataset where the reference maybe wrong. This project is quite large and contains unfortunately a lot of Simulation in multiple folders and subfolders and the same goes for the data sets. I tried to watch the status bar on the lower left in the PK-sim window. There it looks like the simulation are called one after another for creating the jason file after I pushed the button to export the snapshot. But not the full names are shown but just the first 12 digits or so. As the projects names are often the same in the beginning, that was not successful. It would be great to get the full simulation name before it throws the error.

Best, Tobias
